### PR TITLE
Add options to run-test.sh to generate code coverage reports for coreclr

### PR DESCRIPTION
**Pull request by Sridhar-MS**
Changes in run-test.sh to download the unix code-coverage report generator tools from myget package 'unix-code-coverage-tools.1.0.0.nupkg' and use them to generate html graphic code-coverage reports for coreclr.

Sample Code Coverage Summary:
screen shot 2015-09-10 at 6 09 54 pm

screen shot 2015-09-10 at 6 12 54 pm

### CODE REVIEW

**Comment by Sridhar-MS** [comment]
cc: @mmitche

**Comment by stephentoub** [comment]
Thanks, @Sridhar-MS.  Why is this in corefx?  Is the idea that we get a coreclr cc report when running corefx tests?

**Comment by Sridhar-MS** [comment]
Yes @stephentoub. There is not an official way of running [coreclr tests](https://github.com/dotnet/coreclr/tree/master/tests) on linux or osx. Once we fix that we can add similar options to create code coverage reports when running coreclr tests. 

For now we will use corefx tests to get cc reports for linux and osx.

File: run-test.sh [hunk 1]
```diff
@@ -211,6 +219,50 @@ runtest()
...
+  mkdir -p $reportsDir
+  pushd $toolsDir > /dev/null
+
+  echo "Pulling down code coverage tools"
+  wget -q https://www.myget.org/F/dotnet-buildtools/api/v2/package/unix-code-coverage-tools/1.0.0 -O $packageName
+  echo "Unzipping to $toolsDir"
+  unzip -q -o $packageName
+
+  # Invoke gcovr
+  chmod a+rwx ./gcovr
```
Comment (mmitche) [inline run-test.sh]: Do these file permissions not survive the nuget packing?

Comment (davidfowl) [inline run-test.sh]: The only tech I know of that preserves packing is .tar.gz

Comment (mmitche) [inline run-test.sh]: Fine with me then.

**Comment by mmitche** [comment]
LGTM

**Pull request merged**